applications/luci-app-aria2/root/usr/share/rpcd/acl.d/luci-app-aria2.json

```diff
@@ -2,6 +2,18 @@
 	"luci-app-aria2": {
 		"description": "Grant UCI access for luci-app-aria2",
 		"read": {
+			"ubus": {
+				"service": [ "list" ]
+			},
+			"file": {
+				"/etc/group": [ "read" ],
+				"/etc/passwd": [ "read" ],
+				"/bin/cat": [ "exec" ],
```

Comment:
Granting full access to cat and grep will allow read access to any file on the system, this is an undesired privilege escalation.

You shouldn’t need cat, can be replaced with fs.read() and corresponding file read ACLs. As for the grep, I believe you only use it to check for installed webui files. What about probing well known html or js files with fs.stat() instead? This way you can eliminate the need for grep as well.

If you absolutely need this grep functionality, you need to move it into a helper executable which only allows grepping inside the intended target file and not, say, /etc/config/wireless by injecting further command arguments

Reply:
> Granting full access to cat and grep will allow read access to any file on the system, this is an undesired privilege escalation.

It seems that the privilege escalation is inevitable unless we drop the features of reading log from the file and displaying the current config file. Because these files can be configurable. We can't grant LuCI the read access for every possible files.


> As for the grep, I believe you only use it to check for installed webui files. What about probing well known html or js files with fs.stat() instead? This way you can eliminate the need for grep as well.

OK.